### ISSUE TITLE
in multi languages use case, If the default language does not exist, other languages are not applied

### ISSUE THREAD

**Issue by tocgic** [LikeTheSalad/android-stem#21]
**version 2.0.0**

I am really grateful for what "template_" has been removed from 1.3.0 to 2.0.0. 🙇‍♂️ 

It seems to be really good when I go to one of 2.0.0, but it becomes more processing. 🙏 

When using multi language,
If the MAIN RES ID does not use $ {app_name}, it will not be reflected in other languages.

Within our app/`main/res/values` folder, we have the following file:
```xml
<!--strings.xml-->
<resources>
    <string name="app_name">Test</string>
    <string name="welcome_message">Welcome!</string>
</resources>
```
👉   default ID 'welcome_message' is **not used ${app_name}**

Then, Within our app/main/res/values`-es` folder, we have the following file:
```xml
<!--any_file.xml-->
<resources>
    <string name="welcome_message">Bienvenido a ${app_name}</string>
</resources>
```
👉   es ID 'welcome_message' is **used ${app_name}**

👉   In this case, `es string` expects "Bienvenido a Test", but it is `Bienvenido a ${app_name}`

**Comment by LikeTheSalad** [comment]
Hi @tocgic, thanks for your input.

I'm not sure if this is something that should be fixed, or if it should instead be some configurable feature, because the thing is that, usually the "main" and "language-specific" strings should have the same structure, meaning that if you need to use a placeholder in the language-specific one, it's because that placeholder is in the main one too. This helps finding placeholders quicker by just checking the main strings instead of having to go through all the languages too. However, I know that each project is different and can have its own unique requirements, I'm just trying to figure out if this requirement could be common across different projects, or rare, which I think is more the case here.

So just out of curiosity, do you believe that this issue where the same String in different languages has different structures is something common? - do you think you could bump into this same issue in multiple projects? Or do you think this is a very special exception that only happens in the project you're working on right now due to some extraordinary circumstances that don't allow you to keep the same structure between the "main" and "language-specific" strings?

**Comment by Florian14** [comment]
We just observed the same behaviour in my current project. For the context, my app is a fork of another app, we changed the default language of the app and edited some strings. We added a placeholder only in one translation but not in the default one (we could have changed it but we didn't because it was not really necessary for us).

This is not really a problem but it took some time for us to observe it as the string was on a secondary screen, if you won't fix it, it would be nice to have a note about it 🙂

**Comment by LikeTheSalad** [comment]
I see, thanks for the context @Florian14, I just wanted to make sure that there were cases in which there's no other way around but to define templates only on translations and not in the main resources, and taking into account that you're not the first person mentioning it, I think we can safely say that it can happen, although it doesn't seem to be a very common case. So for now I will address it as a configurable option that would make the plugin to search through all string resources across all languages if enabled. Now, this will definitely mean some performance penalties for this plugin, as it will have to search across all languages, as opposed to just the main one as it is now, so this new behavior will be disabled by default.

**Comment by LikeTheSalad** [comment]
Hi @tocgic, @Florian14 

I just released the version 2.1.0 of Stem, where you'd be able to turn on the config for searching templates not only through your main string resources, but also through your language-specific ones. It is disabled by default so in order to enable it, you'd need to add this block of code into your `build.gradle` file where you've applied Stem plugin into:

```groovy
androidStem {
    includeLocalizedOnlyTemplates = true
}
```

More info on the [README file](https://github.com/LikeTheSalad/android-stem#3--configuration).

Please feel free to reopen this ticket if you're still seeing the same issue after turning the config on.

Cheers!
